### PR TITLE
[porn91] Add new extractor for 91porn.com

### PR DESCRIPTION
**Pull request by PeterDing**
An extractor for 91porn.com -- A popular Chinese amateur porn site.

[WARN] 91porn.com has a limit for downloading, one ip could download 10 videos among 24h.

-----

This one will be good.

### CODE REVIEW

**Comment by yan12125** [comment]
FYI: [```_search_regex```](https://github.com/rg3/youtube-dl/blob/master/youtube_dl/extractor/common.py#L526)/[```_html_search_regex```](https://github.com/rg3/youtube-dl/blob/master/youtube_dl/extractor/common.py#L560) is more robust than ```re.search```. Please consider using these helper functions.

File: youtube_dl/extractor/porn91.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+        title = title.replace('\n', '')
+
+        # get real url
+        n1 = self._search_regex(
+            r'so.addVariable\(\'file\',\'(?P<n1>\d+)\'', webpage, 'n1')
+        n2 = self._search_regex(
+            r'so.addVariable\(\'seccode\',\'(?P<n2>.+?)\'', webpage, 'n2')
+        n3 = self._search_regex(
+            r'so.addVariable\(\'max_vid\',\'(?P<n3>\d+)\'', webpage, 'n3')
+        if not (n1 and n2 and n3):
```
Comment (yan12125) [inline youtube_dl/extractor/porn91.py]: ```_search_regex``` raises a ```RegexNotFoundError``` if the given pattern is not found. As a result, this checking does not work.

File: youtube_dl/extractor/porn91.py [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    def _real_extract(self, url):
+        mobj = re.match(self._VALID_URL, url)
+        video_id = mobj.group('id')
+        url = 'http://91porn.com/view_video.php?viewkey=%s' % video_id
+        self._set_cookie('91porn.com', 'language', 'cn_CN')
+        webpage = self._download_webpage(url, video_id, "get HTML content")
+        title = self._search_regex(
+            r'<div id="viewvideo-title">(?P<title>.+?)</div>',
+            webpage, 'title', flags=re.DOTALL)
+        assert title
```
Comment (yan12125) [inline youtube_dl/extractor/porn91.py]: It's not necessary because ```_search_regex``` already raises exceptions in failure.

**Comment by yan12125** [comment]
Thanks for the extractor. Merged with some modifications.

**Comment by yan12125** [comment]
Would you like to be listed in [AUTHORS](https://github.com/rg3/youtube-dl/blob/master/AUTHORS)? If so, under what name?

**Comment by PeterDing** [comment]
@yan12125 It is honor, you can call me Peter Ding.

**Comment by yan12125** [comment]
Thanks. Done in 4053ee9104cd7669f749a267dc2c2a1725ca188b

**Pull request merged**